woocommerce/woocommerce-gateway-stripe
# Fix onClose event handler for Blocks

Specifically, in Block Checkout, the `onClose` handler was not being fired when dismissing the payment request modal. I believe this was in part due to how event handlers were removed when a hook is unmounted using `removeAllListeners` which removes all events. Since we have multiple `useEffects` all at once, I think there was some kind of race condition.

I've fixed this by removing specific events on unmount rather than all of them.

# Testing instructions

On block checkout, pay using the Stripe Express Checkout but cancel the payment request. onClose handler should fire. This is easier to verify when using https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4228 since the loading state of the express payment methods will change once closed. Without this fix, the loading state will be stuck.

-------------------
